rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -282,6 +360,92 @@ trait NodeInfoServiceCached extends NodeInfoService  with Loggable with CachedRe
   }
 
   /**
+   * An utility method that gets data from backend for things that are
+   * node really nodes (pending or deleted).
+   */
+  private[this] def getNotAcceptedNodeDataFromBackend(status: InventoryStatus): Box[Map[NodeId, NodeInfo]] = {
```

Comment:
Almost all code here is the same as accepted Node :( I would be very happuy if they were merged ...